### PR TITLE
feature: add costTime of autoTest plan list

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
feature: add costTime of autoTest plan list

#### Which issue(s) this PR fixes:
https://erda.cloud/erda/dop/projects/387/issues/all?id=222572&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMTI2MSJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=506&type=TASK
- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

#### Need cherry-pick to release versions?
/cherry-pick release/1.3

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/openapi/component-protocol/scenarios/auto-test-plan-detail/components/executeTaskTable/render.go [hunk 1]
```diff
@@ -488,6 +499,28 @@ func (a *ExecuteTaskTable) setData(pipeline *apistructs.PipelineDetailDTO) error
...
+		return "-"
+	}
+	if !task.IsSnippet {
+		return time.Unix(task.CostTimeSec, 0).In(time.UTC).Format("15:04:05")
+	}
+	if task.SnippetPipelineID == nil {
+		return "-"
+	}
+
+	rsp, err := a.CtxBdl.Bdl.GetPipeline(*task.SnippetPipelineID)
```
Comment (sfwn) [inline modules/openapi/component-protocol/scenarios/auto-test-plan-detail/components/executeTaskTable/render.go:517]: snippet task can also use task's costTimeSec.

Comment (sfwn) [reply modules/openapi/component-protocol/scenarios/auto-test-plan-detail/components/executeTaskTable/render.go:517]: You have to care about performance issue.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/openapi/component-protocol/scenarios/auto-test-plan-detail/components/executeTaskTable/render.go (hunk 1 above)
Comment (littlejiancc) [reply modules/openapi/component-protocol/scenarios/auto-test-plan-detail/components/executeTaskTable/render.go:517]: the costTimeSec  of snippet task is always -1

Comment (littlejiancc) [reply modules/openapi/component-protocol/scenarios/auto-test-plan-detail/components/executeTaskTable/render.go]: now update snippet pipeline task time when reconciler done

File: modules/pipeline/dbclient/op_pipeline_task.go [hunk 2]
```diff
@@ -189,6 +189,27 @@ func (client *Client) UpdatePipelineTaskStatus(id uint64, status apistructs.Pipe
 	return err
 }
 
+func (client *Client) UpdatePipelineTaskTime(p *spec.Pipeline, ops ...SessionOption) error {
```
Comment (sfwn) [inline modules/pipeline/dbclient/op_pipeline_task.go:192]: Add comment for exported function.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
Add screenshot of snippet task time related data.

**Comment by littlejiancc** [comment]
> Add screenshot of snippet task time related data.

image

**Review by sfwn — APPROVED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**